### PR TITLE
Fix incorrect assignment of layout as a data source (Issue #992)

### PR DESCRIPTION
**Pull request by willium**
Fixes bug in #992 where non-ordinal marks are incorrectly set to use layout as a data source in the root group

### CODE REVIEW

File: test/compile/compile.test.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        "encoding": {
+          "x": {"field": "Horsepower","type": "quantitative"},
+          "y": {"field": "Miles_per_Gallon","type": "quantitative"},
+          "color": {"field": "Origin", "type": "ordinal"}
+        }
+      });
+
+      const rootGroup = compile.compileRootGroup(model);
+
+      it('should not have layout dataref', function() {
```
Comment (kanitw) [inline test/compile/compile.test.ts]: very minor:  maybe  "should refer to layout data" ?

File: test/compile/compile.test.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        "mark": "bar",
+        "encoding": {
+          "x": {"field": "a", "type": "ordinal"},
+          "y": {"field": "b", "type": "quantitative"}
+        }
+      });
+
+      const rootGroup = compile.compileRootGroup(model);
+
+      it('should have layout dataref', function() {
```
Comment (kanitw) [inline test/compile/compile.test.ts]: Same here (maybe  "should refer to layout data" ?)

**Comment by kanitw** [comment]
LGTM, feel free to merge after addressing minor comments.

File: test/compile/compile.test.ts (hunk 1 above)
Comment (willium) [inline test/compile/compile.test.ts]: done.

**Comment by willium** [comment]
These tests could easily be more thorough and theres other things to test in compile (it's probably a good idea to do), but I didn't want to chase down every edge case when we're fairly sure it works and this is relatively harmless compared to other potential tests. We can improve coverage later?

File: test/compile/compile.test.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+    describe('non-ordinal', function() {
+      const model = parseModel({
+        "description": "A scatterplot showing horsepower and miles per gallons.",
+        "data": {"url": "data/cars.json"},
+        "mark": "point",
+        "encoding": {
+          "x": {"field": "Horsepower","type": "quantitative"},
+          "y": {"field": "Miles_per_Gallon","type": "quantitative"},
+          "color": {"field": "Origin", "type": "ordinal"}
```
Comment (domoritz) [inline test/compile/compile.test.ts]: color channel is not needed, right?

**Comment by domoritz** [comment]
Yeah, I think that's fine. To make this function more testable, we should probably extract the lines that get the width and height anyway. but let's defer that.

**Comment by kanitw** [comment]
Yeah, don't test the top-level compile too much.   This will change vastly in the upcoming month to support composition.

File: test/compile/compile.test.ts (hunk 3 above)
Comment (willium) [inline test/compile/compile.test.ts]: removed and rebased
